### PR TITLE
Erdikose/feat/minimal cluster dashboard

### PR DESCRIPTION
**Pull request by erdkse**
This PR...

## Changes

-

## Fixes

-

## How to test it

-

## Screenshots

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/organisms/Dashboard/Header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import {TitleBar} from '@monokle/components';
+
+import * as S from './Header.styled';
+
+export const Header = ({title}: {title: string}) => {
+  const dispatch = useAppDispatch();
+  const resourceMap = useAppSelector(state => state.main.resourceMap);
+  const selectedNamespace = useAppSelector(state => state.dashboard.ui.selectedNamespace);
+
+  const getNamespaces = (): K8sResource[] => {
```
Comment (topliceanurazvan) [inline src/components/organisms/Dashboard/Header/Header.tsx:17]: Can we wrap this in a useCallback ?

File: src/components/organisms/Dashboard/Overview/Overview.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+import {InventoryInfo} from './InventoryInfo';
+import * as S from './Overview.styled';
+import {Status} from './Status';
+import {Utilization} from './Utilization';
+
+export const Overview = () => {
+  const [clusterInformation, setClusterInformation] = useState<ClusterInformation | null>(null);
+  const [activityData, setActivityData] = useState<ClusterEvent[]>([]);
+  const [utilizationData, setUtilizationData] = useState<NodeMetric[]>([]);
+  const [hearbeat, setHearbeat] = useState(0);
```
Comment (topliceanurazvan) [inline src/components/organisms/Dashboard/Overview/Overview.tsx:33]: Should this be `heartbeat` ?

File: src/components/organisms/Dashboard/Overview/Status.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  const resourceMap = useAppSelector(state => state.main.resourceMap);
+  const selectedNamespace = useAppSelector(state => state.dashboard.ui.selectedNamespace);
+
+  const getResourceCount = useCallback(() => {
+    return Object.values(resourceMap).filter(r =>
+      selectedNamespace !== 'ALL' ? selectedNamespace === r.namespace : true
+    ).length;
+  }, [resourceMap, selectedNamespace]);
+
+  const getErrorCount = () => {
```
Comment (topliceanurazvan) [inline src/components/organisms/Dashboard/Overview/Status.tsx:19]: Can we have useCallback here?

File: src/components/organisms/Dashboard/Overview/Status.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    return Object.values(resourceMap)
+      .filter(resource => (selectedNamespace !== 'ALL' ? selectedNamespace === resource.namespace : true))
+      .reduce(
+        (total: number, resource: K8sResource) =>
+          total + (resource.validation && resource.validation.errors ? resource.validation.errors.length : 0),
+        0
+      );
+  };
+
+  const getWarningCount = () => {
```
Comment (topliceanurazvan) [inline src/components/organisms/Dashboard/Overview/Status.tsx:29]: Can we have useCallback here?

File: src/components/organisms/Dashboard/Tableview/InfoTab.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+import {K8sResource} from '@models/k8sresource';
+
+import {useAppSelector} from '@redux/hooks';
+
+import {timeAgo} from '@utils/timeAgo';
+
+import * as S from './InfoTab.styled';
+
+export const InfoTab = ({resourceId}: {resourceId: string}) => {
+  const resourceMap = useAppSelector(state => state.main.resourceMap);
```
Comment (topliceanurazvan) [inline src/components/organisms/Dashboard/Tableview/InfoTab.tsx:14]: Can we directly return from here the resource, without needing an extra useEffect and useState ? 

```
useAppSelector(state => state.main.resourceMap[resourceId]);
```

File: src/components/organisms/DashboardPane/DashboardPane.tsx [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+          }
+          return output;
+        },
+        {Overview: {}}
+      )
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [getRegisteredKindHandlers(), activeMenu, leftMenu]);
+
+  const setActiveMenu = (section: string) => {
```
Comment (topliceanurazvan) [inline src/components/organisms/DashboardPane/DashboardPane.tsx:66]: Can we have this as useCallback?

File: src/components/organisms/DashboardPane/DashboardPane.tsx [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+  const getResourceCount = useCallback(
+    (kind: string) => {
+      return Object.values(resourceMap).filter(
+        r => r.kind === kind && (selectedNamespace !== 'ALL' ? selectedNamespace === r.namespace : true)
+      ).length;
+    },
+    [resourceMap, selectedNamespace]
+  );
+
+  const getErrorCount = (kind: string) => {
```
Comment (topliceanurazvan) [inline src/components/organisms/DashboardPane/DashboardPane.tsx:80]: same as above

File: src/components/organisms/DashboardPane/DashboardPane.tsx [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+          resource.kind === kind && (selectedNamespace !== 'ALL' ? selectedNamespace === resource.namespace : true)
+      )
+      .reduce(
+        (total: number, resource: K8sResource) =>
+          total + (resource.validation && resource.validation.errors ? resource.validation.errors.length : 0),
+        0
+      );
+  };
+
+  const getWarningCount = (kind: string) => {
```
Comment (topliceanurazvan) [inline src/components/organisms/DashboardPane/DashboardPane.tsx:93]: same as above

**Review by topliceanurazvan — CHANGES_REQUESTED** [review]

**Review by devcatalin — APPROVED** [review]

**Review by topliceanurazvan — APPROVED** [review]

**Pull request merged**